i3_agenda/event_utils.py

```diff
@@ -0,0 +1,49 @@
+
+import time
+import datetime
+from typing import Optional, List
+
+from event import Event
+
+
+def get_closest(events: List[Event], hide_event_after: int) -> Optional[Event]:
+    closest = None
+    for event in events:
+        # Don't show all day events
+        if event.is_allday:
+            continue
+
+        now = time.time()
+        # If the event already ended
+        if event.end_time < now:
+            continue
+
+        # If the event started more than hide_event_after ago
+        if hide_event_after > -1 and event.unix_time + 60 * hide_event_after < now:
+            continue
+
+        if closest is None or event.unix_time < closest.unix_time:
+            closest = event
+
+    return closest
+
+
+def get_event_time(full_time: str) -> float:
```

Comment:
this function could be a method of the `Event` class

Reply:
Well.. I thinking about this, it gets a string of the event time and returns the unix time of this string, and then this unix time is used to create the Event object. So I don't see it as a Event method, do you agree with me? Maybe the name of this function can be changed to something like unix_time_from_string.

Reply:
Yeah you're right, I have a different suggestion that I think could solve several issues:

Add another constructor to the `Event` object, that will receive the json Event object (the original one, the one that is used by `event_from_json`).  
This constructor will do the parsing that is currently done by the `event_from_json`. It will create an `Event` object after doing the parsing using the code from `event_from_json`.  
This way, events could be loaded correctly from cache, like they used to, and also could be created by passing an event json (received from Google). This will solve this issue, and also the "is allday" issue, since it could all be done in the constructor.

Let me know if you understood what I mean

Reply:
Yeah.. I realy liked this approach.

Reply:
I agree, the second option is better.  
Have a static function in event.py that takes an Event json and constructs an Event object